src/transport/AdminPairingTable.cpp

```diff
@@ -28,6 +28,18 @@ namespace chip {
 
 namespace Transport {
 
+CHIP_ERROR AdminPairingInfo::SetFabricLabel(uint8_t * fabricLabel)
+{
+    char *fabricLabelChar = Uint8::to_char(fabricLabel);
+    if (strlen(fabricLabelChar) + 1 > 32)
+    {
+        ChipLogError(Discovery, "Cannot set fabricLabel as it exceeds allowed size of 32.");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    memmove(mFabricLabel, fabricLabelChar , strlen(fabricLabelChar) + 1);
```

Comment:
Compute length once with validation and reuse, rather than re-call strlen.

Also, do not expect to copy a nul-terminator, but rather forcibly set one past the actual length.

BTW why `memmove` instead of `memcpy`?